### DISCUSSION TITLE
Mesh Network Functionality

### DISCUSSION THREAD

**Discussion by Winterhuman** [alecbcs/hyprspace#35] [Q&A] [answered]
Just a question on how hyprspace works:

Let's say we have Peers A, B, and C. The connections initially starts as follows:

Peer B <-> Peer A
Peer C <-> Peer A

If Peer C tries to ping Peer B, does the ping request go through Peer A or does Peer A tell Peer C about Peer B's multiaddresses and ping Peer B directly?

**Comment by Winterhuman** [comment, marked as answer]
Nevermind, just realised defining Peer B and C in the peer list alongside Peer A would achieve a mesh network as described here, does make me wonder if some form of dynamic peer list could be introduced to allow peers to tell each other about themselves.

**Reply by alecbcs** [reply]
> ...does make me wonder if some form of dynamic peer list could be introduced to allow peers to tell each other about themselves.

It's on my list! (The ability to share/send a peer-config to another peer over libp2p.)

**Reply by Winterhuman** [reply]
Oh nice, looking forward to seeing it!

**Reply by bmullan** [reply]
@alecbcs 
Yes this is a sorely needed feature and without it updating (expanding or contracting) the network config of many Peers is.
